### PR TITLE
Disabling automatic carousel transitions on user interaction

### PR DESCRIPTION
**Pull request by ouchadam**
Part of #4584
 
- Disables the automatic carousel transitions once the user interacts with the carousel

| BEFORE | AFTER |
| --- | --- |
before-carousel|after-carousel

### CODE REVIEW

File: vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthSplashCarouselFragment.kt [hunk 1]
```diff
@@ -80,17 +80,27 @@ class FtueAuthSplashCarouselFragment @Inject constructor(
...
-                override fun onPageSelected(position: Int) {
-                    scheduledTransition?.cancel()
+            override fun onPageScrolled(position: Int, positionOffset: Float, positionOffsetPixels: Int) {
+                hasUserManuallyInteractedWithCarousel = !isFakeDragging
+            }
+
+            override fun onPageSelected(position: Int) {
+                scheduledTransition?.cancel()
+                if (hasUserManuallyInteractedWithCarousel) {
+                    // stop the automatic transitions
```
Comment (bmarty) [inline vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthSplashCarouselFragment.kt:98]: This is fine but maybe avoid having empty block. I think lint/compiler will also complain. Better to write `if(!...)` and just add the comment above.

**Review by bmarty — APPROVED** [review]
Thanks! Just one small remark

File: vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthSplashCarouselFragment.kt (hunk 1 above)
Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthSplashCarouselFragment.kt:98]: in this case the comment counts as a non empty block, will invert the condition and remove the branch

Comment (ouchadam) [reply vector/src/main/java/im/vector/app/features/onboarding/ftueauth/FtueAuthSplashCarouselFragment.kt]: 01ef46517e0358d0446a9b13fd0612e6286c4883

**Review by bmarty — APPROVED** [review]

**Pull request merged**